test/scale-infra/deploy_ks_core.yaml

```diff
@@ -0,0 +1,124 @@
+      shell: |
+             kubectl config rename-context kubernetes-admin@kubernetes kscore --ignore-not-found
+             helm delete ks-core  --ignore-not-found
+             helm -n open-cluster-management delete status-addon  --ignore-not-found
+             kubectl --context kscore delete ns open-cluster-management-hub open-cluster-management  --ignore-not-found
+             kubectl config unset contexts.wds1 --ignore-not-found
+             kubectl --context kscore delete -f https://raw.githubusercontent.com/kubestellar/kubestellar/refs/heads/release-"{{ ks_release }}"/example/kind-nginx-ingress-with-SSL-passthrough.yaml  --ignore-not-found
+      when: result is success
+
+    - name: install KubeStellar using core helm-cart
+      shell: |
+             export KUBESTELLAR_VERSION="{{ ks_release }}"
+             helm upgrade --install ks-core oci://ghcr.io/kubestellar/kubestellar/core-chart \
+             --version $KUBESTELLAR_VERSION \
+             --set-json='ITSes=[{"name":"its1","type":"host"}]' \
+             --set-json='WDSes=[{"name":"wds1"}]'
+
+    - name: Get kubeconfig "contexts" for wds1 and its1 spaces
+      shell: |
+             kflex ctx wds1
+
+    - name: install ingress controller
+      shell: |
+             kubectl config rename-context kubernetes-admin@kubernetes kscore
+             kubectl --context kscore label node --all ingress-ready="true"
+             kubectl --context kscore apply -f https://raw.githubusercontent.com/kubestellar/kubestellar/refs/heads/release-0.24.0/example/kind-nginx-ingress-with-SSL-passthrough.yaml
+
+    - name: update wds1 space with the correct port for the ingress controller
+      shell: kubectl --context kscore -n ingress-nginx get svc ingress-nginx-controller -o=jsonpath='{.spec.ports[1].nodePort}'
+      register: ingress_port
+
+    # - name: Debugging message
+    #   ansible.builtin.debug:
+    #    msg: "{{ ingress_port.stdout }}"
+
+    - name: Replace old hostname with new hostname (requires Ansible >= 2.4)
+      ansible.builtin.replace:
+        path: /root/.kube/config
+        regexp: '(\s+)https://wds1\.localtest\.me:9443(\s+.*)?$'
+        replace: '\1https://wds1.localtest.me:{{ ingress_port.stdout }}\2'
+
+
+    - name: Sleep for 30 seconds and continue with play
+      ansible.builtin.wait_for:
+        timeout: 30
+      delegate_to: localhost
+
+    - name: Get the token to register managed clusters
+      shell: clusteradm --context kscore get token | grep '^clusteradm join'
+      register: wec_join_command
+      until: "wec_join_command is not failed"
+      retries: 3
+      delay: 10
+    
+    - name: Set some environment configurations for the end user
+      shell: |
+             alias k="kubectl"
+             k config use-context kscore
+             
+    - name: Copy join command to local file (Ansible control node)
+      local_action: copy content="{{ wec_join_command.stdout }}" dest="/tmp/wec_join_command" mode=0777
```

Comment:
Is one token good for multiple WECs?

Reply:
Yes! More precisely, this token is later copied to all WEC instances, where the appropriated WEC name info is replaced.